### PR TITLE
restrict anonymous access to user information update

### PR DESCRIPTION
**Pull request by tino097**
Related to #3915 with small updates

### CODE REVIEW

File: ckan/logic/action/get.py [hunk 1]
```diff
@@ -1184,7 +1184,10 @@ def _group_or_org_show(context, data_dict, is_org=False):
         packages_field = None
 
     include_tags = asbool(data_dict.get('include_tags', True))
-    include_users = asbool(data_dict.get('include_users', True))
+    if asbool(config.get('ckan.auth.public_user_details', True)):
+        include_users = asbool(data_dict.get('include_users', True))
+    else:
+        include_users = asbool(data_dict.get('include_users', False))
```
Comment (smotornyuk) [inline ckan/logic/action/get.py]: It's not a part of this feature, but maybe it's a time to fix *ValueError*, raised when asbool receives an incorrect value. Maybe, wrap this block with try/except for converting it into ValidationError.  If you try to get **/api/action/group_show?id=GROUP_ID&include_users=tr**, you'll see what i mean

File: ckan/logic/auth/get.py [hunk 2]
```diff
@@ -178,14 +195,19 @@ def vocabulary_show(context, data_dict):
...
 
+
 def user_show(context, data_dict):
     # By default, user details can be read by anyone, but some properties like
     # the API key are stripped at the action level if not not logged in.
-    return {'success': True}
+    if not asbool(config.get('ckan.auth.public_user_details', True)):
+        return restrict_anon(context)
+    else:
+        return {'success': True}
```
Comment (smotornyuk) [inline ckan/logic/auth/get.py]: Do we have any 403 handlers for flask application? After disabling public_users, visiting '/user' page results in base **Forbidden** page. I just don't remember, whether it's ok and we are going to implement error handlers separately or they already are implemented

File: ckan/logic/action/get.py [hunk 3]
```diff
@@ -1261,7 +1264,7 @@ def group_show(context, data_dict):
          (optional, default: ``True``)
     :type include_groups: bool
     :param include_tags: include the group's tags
-         (optional, default: ``True``)
+         (optional, default: ``False``)
```
Comment (smotornyuk) [inline ckan/logic/action/get.py]: https://github.com/ckan/ckan/pull/4215/files#diff-4494aaf212251212949348df94158668R1186 - it still set to true

File: ckan/logic/auth/get.py (hunk 2 above)
Comment (tino097) [reply ckan/logic/auth/get.py]: lets discuss at the meeting later

File: ckan/logic/action/get.py (hunk 3 above)
Comment (tino097) [reply ckan/logic/action/get.py]: i guess its by mistake, didnt notice  it when i was doing the changes. Thanks!

**Comment by amercader** [comment]
@smotornyuk can you have a final look and merge if happy? If not let me know

File: ckan/logic/action/get.py [hunk 4]
```diff
@@ -1255,7 +1261,7 @@ def group_show(context, data_dict):
          (optional, default: ``True``)
     :type include_extras: bool
     :param include_users: include the group's users
-         (optional, default: ``True``)
+         (optional, default: ``False``)
```
[the remark this answers is not in the document]
Comment (tino097) [reply ckan/logic/action/get.py]: yeah, it should be `True`

**Comment by amercader** [comment]
Merged in 0ea7752fa72693a8ab5df332cc94fa700a6747ee

**Pull request closed without merging**